cmd/local/apicollect/jobprofiles.go

```diff
@@ -35,39 +35,42 @@ import (
 func GetNumberOfJobProfilesCollected(c *conf.CollectConf) (tried, collected int, err error) {
 	var files []fs.DirEntry
 	var queriesrows []queriesjson.QueriesRow
-	if !c.IsDremioCloud() {
-		files, err = os.ReadDir(c.QueriesOutDir())
-		if err != nil {
-			return 0, 0, err
-		}
-		queriesjsons := []string{}
-		for _, file := range files {
-			queriesjsons = append(queriesjsons, path.Join(c.QueriesOutDir(), file.Name()))
-		}
 
-		if len(queriesjsons) == 0 {
-			simplelog.Warning("no queries.json files found. This is probably an executor, so we are skipping collection of Job Profiles")
-			return
+	files, err = os.ReadDir(c.SystemTablesOutDir())
+	if err != nil {
+		return 0, 0, err
+	}
+	jobhistoryjsons := []string{}
+	for _, file := range files {
+		if strings.Contains(file.Name(), "project.history.jobs") || strings.Contains(file.Name(), "jobs_recent") {
+			jobhistoryjsons = append(jobhistoryjsons, path.Join(c.SystemTablesOutDir(), file.Name()))
 		}
+	}
 
-		queriesrows = queriesjson.CollectQueriesJSON(queriesjsons)
-	} else {
-		files, err = os.ReadDir(c.SystemTablesOutDir())
-		if err != nil {
-			return 0, 0, err
-		}
-		jobhistoryjsons := []string{}
-		for _, file := range files {
-			if strings.Contains(file.Name(), "project.history.jobs") {
-				jobhistoryjsons = append(jobhistoryjsons, path.Join(c.SystemTablesOutDir(), file.Name()))
+	if len(jobhistoryjsons) == 0 {
+
+		// Attempt to read job history from queries.json, if not Dremio Cloud
+		if !c.IsDremioCloud() {
```

Comment:
stupid out loud question..is this even necessary now..basically len(jobhistoryjsons) == 0 will fail consistently for dremiocloud no? Like effectively we will never parse in practice

Reply:
also have a failing test 

--- FAIL: TestGetNumberOfJobProfilesCollectedWIthServerUp (0.28s)
    jobprofiles_test.go:98: failed running job profile numbers generation
        open /tmp/TestGetNumberOfJobProfilesCollectedWIthServerUp2146487464/002/system-tables/node1: no such file or directory

Reply:
The logic now is:
1. Look for Job JSONs that are either from `sys.jobs_recent` (Software) or `sys.project.history.jobs` (Cloud) and collect those job infos
2. If no valid job entries are found, either a) look for queries.json instead (Software) or b) skip job profile collection (Cloud)
2a. If neither Job JSONs or queries.json are found, skip job profile collection (Software)